packages/timeline-state-resolver/src/service/device.ts

```diff
@@ -78,19 +91,28 @@ export abstract class Device<DeviceOptions, DeviceState, Command extends Command
 export interface BaseDeviceAPI<DeviceState, Command extends CommandWithContext> {
 	/**
 	 * This method takes in a Timeline State that describes a point
-	 * in time on the timeline and returns a decice state that
-	 * describes how the device should be according to the timeline state
-	 *
+	 * in time on the timeline and converts it into a "device state" that
+	 * describes how the device should be according to the timeline state.
+	 * This is optional, and intended to simplify diffing logic in
```

Comment:
It's not optional, optional would mean the developer can omit the method entirely. Whether the implementation transforms the state at all is up to the developer though